graylog2-server/src/main/java/org/graylog/security/DBGrantService.java

```diff
@@ -89,10 +89,8 @@ public DBGrantService(MongoConnection mongoConnection,
                 DBQuery.in(GrantDTO.FIELD_GRANTEE, grantees))).toArray();
     }
 
-    public GrantDTO create(GrantDTO grantDTO, @Nullable User currentUser) {
+    public GrantDTO create(GrantDTO grantDTO, String userName) {
         final ZonedDateTime now = ZonedDateTime.now(ZoneOffset.UTC);
-        final String userName = requireNonNull(currentUser, "currentUser cannot be null").getName();
```

Comment:
Since we have a `String` now, I think we should check that the `userName` is not null **and** not empty.